master: nightly linux-no-cgo failed

The nightly linux-no-cgo test run failed on 54bb06f31c01968e39dc84877d93524d6714f7de. Please review the run [here](https://github.com/cockroachdb/pebble/actions/runs/15776487193).

Jira issue: PEBBLE-488

```
--- FAIL: TestCompaction (0.71s)
    compaction_test.go:867: seed: 1750413937850093324
    --- FAIL: TestCompaction/value_separation (0.01s)
        compaction_test.go:945: 
              testdata/compaction/value_separation:137:
              metrics
            output didn't match expected:
              @@ -13,[18](https://github.com/cockroachdb/pebble/actions/runs/15776487193/job/44472033513#step:4:19) +13,18 @@
               WAL: 1 files (0B)  in: 30B  written: 41B (37% overhead)
               Flushes: 1
               Compactions: 4  estimated debt: 0B  in progress: 0 (0B)  canceled: 0 (0B)  failed: 0  problem spans: 0
                            default: 1  delete: 0  elision: 0  move: 3  read: 0  tombstone-density: 0  rewrite: 0  copy: 0  multi-level: 2
               MemTables: 1 ([25](https://github.com/cockroachdb/pebble/actions/runs/15776487193/job/44472033513#step:4:26)6KB)  zombie: 1 (256KB)
              -Zombie tables: 0 (0B, local: 0B)
              +Zombie tables: 1 (844B, local: 844B)
               Backing tables: 0 (0B)
               Virtual tables: 0 (0B)
               Local tables size: 864B
               Compression types: snappy: 1
              -Table stats: all loaded
              -Block cache: 4 entries (1.6KB)  hit rate: 70.3%
              -File cache: 1 tables, 1 blobfiles (376B)  hit rate: 82.2%
              +Table stats: initial load in progress
              +Block cache: 12 entries (4.5KB)  hit rate: 70.3%
              +File cache: 2 tables, 4 blobfiles (944B)  hit rate: 82.2%
               Range key sets: 0  Tombstones: 0  Total missized tombstones encountered: 0
               Snapshots: 0  earliest seq num: 0
               Table iters: 0
               Filter utility: 0.0%
               Ingestions: 0  as flushable: 0 (0B in 0 tables)
```

--- a/compaction_test.go
+++ b/compaction_test.go
@@ -1068,6 +1068,9 @@ func TestCompaction(t *testing.T) {
 				return runLSMCmd(td, d)
 
 			case "metrics":
+				d.mu.Lock()
+				d.waitTableStatsInitialLoad()
+				d.mu.Unlock()
 				m := d.Metrics()
 				return m.StringForTests()
 

